### PR TITLE
evm: Add next block baseFeePerGas in `eth_feeHistory` output

### PR DESCRIPTION
**Pull request by shohamc1**
Refining rpc `eth_feeHistory` with
  - [x] add next block base fee
  - [x] change oldest block type from hash to number
  - [x] some percentiles to fine tune

### CODE REVIEW

File: lib/ain-evm/src/block.rs [hunk 1]
```diff
@@ -246,7 +246,25 @@ impl BlockService {
             Some(reward)
         };
 
+        // add another entry for baseFeePerGas
+        let next_block_base_fee = match self
+            .storage
+            .get_block_by_number(&(first_block + U256::one()))?
```
Comment (canonbrother) [inline lib/ain-evm/src/block.rs:252]: can you test if the first_block == latest_block?

**Review by canonbrother — CHANGES_REQUESTED** [review]

File: test/functional/feature_evm_rpc_fee_history.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+       node = self.nodes[0]
+       for x in range(count):
+            for y in priority_fees:
+                tx = {
+                    "from": self.ethAddress,
+                    "value": "0x0",
+                    "data": CONTRACT_BYTECODE,
+                    "gas": "0x18e70",  # 102_000
+                    "maxPriorityFeePerGas": hex(16),
+                    # "maxPriorityFeePerGas": hex(y), # <-- too-many-eth-txs-by-sender
```
Comment (canonbrother) [inline test/functional/feature_evm_rpc_fee_history.py:108]: edge case

File: lib/ain-evm/src/block.rs [hunk 3]
```diff
@@ -174,6 +174,7 @@ impl BlockService {
             block_number -= U256::one();
         }
 
+        // TODO(): b.hash -> b.number
```
Comment (canonbrother) [inline lib/ain-evm/src/block.rs:177]: todo

File: lib/ain-evm/src/block.rs (hunk 1 above)
Comment (canonbrother) [reply lib/ain-evm/src/block.rs:253]: resolved at https://github.com/DeFiCh/ain/pull/2390/commits/e8243fb9d8e2baacf2a72758083a40fb946a7d2c

File: test/functional/feature_evm_rpc_fee_history.py (hunk 2 above)
Comment (canonbrother) [reply test/functional/feature_evm_rpc_fee_history.py:108]: 64 txs is maximum in mempool for the same sender

File: lib/ain-evm/src/block.rs (hunk 3 above)
Comment (shohamc1) [reply lib/ain-evm/src/block.rs]: Good catch, thank you.
